javascript/README.md

```diff
@@ -88,6 +88,21 @@ var length = items.length;
 var name = 'foo';
 ````
 
+## Don't have unused variables/expressions
+```javascript
+// Bad: Many of these variables/expressions are useless
+{} // object - not used
+var newArray = []; // newArray - not used
+['1', '2', '3', '4'].forEach(function(num, index) { // index - not used
```

Comment:
@jansepar There are options to eslint we can use to ignore unused arguments to functions. That seems like it'd be a pragmatic compromise here. 
```
"no-unused-vars": [2, {"vars": "all", "args": "none" }]
```
Reference: https://github.com/eslint/eslint/blob/master/docs/rules/no-unused-vars.md

*Interestingly, I have no idea what the `2` is for in the array. All examples used `2` and the docs don't say what it's for. :confused: